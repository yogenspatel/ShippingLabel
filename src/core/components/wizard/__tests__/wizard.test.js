/* eslint-disable no-undef */

import React from "react";
import ReactDOM from "react-dom";
import Wizard from "../index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Wizard Component renders correctly", () => {
  it("has rendered without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Wizard />, div);
  });
  it("has called snapshot with Wizard component", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn
    }
    const tree = shallow(<Wizard {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("has clicked on next without error", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn
    }
    const wrapper = shallow(<Wizard {...props} />);
    expect(wrapper.find(".next").length).toBe(1);
    wrapper.setState({
        step: 1,
        totalSteps: 5,
        error: false
    });
    wrapper.find(".next").simulate("click");
    expect(wrapper.state('step')).toBe(2);
  });

  it("has clicked on next button with error state true", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn
    }
    const wrapper = shallow(<Wizard {...props} />);
    expect(wrapper.find(".next").length).toBe(1);
    wrapper.setState({
        step: 1,
        totalSteps: 5,
        error: true
    });
    wrapper.find(".next").simulate("click");
    expect(wrapper.state('step')).toBe(1);
  });
  it("has clicked on prev button with error state true", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn
    }
    const wrapper = shallow(<Wizard {...props} />);
    expect(wrapper.find(".prev").length).toBe(1);
    wrapper.setState({
        step: 4,
        totalSteps: 5,
        error: true
    });
    wrapper.find(".prev").simulate("click");
    expect(wrapper.state('step')).toBe(3);
  });
  it("has clicked on prev button", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn
    }
    const wrapper = shallow(<Wizard {...props} />);
    expect(wrapper.find(".prev").length).toBe(1);
    wrapper.setState({
        step: 4,
        totalSteps: 5,
        error: false
    });
    wrapper.find(".prev").simulate("click");
    expect(wrapper.state('step')).toBe(3);
  });
  it("has clicked on next button on the last step", () => {
    const props = {
        steps: [1,2,3,4,5],
        onComplete: jest.fn()
    }
    const wrapper = shallow(<Wizard {...props} />);
    expect(wrapper.find(".next").length).toBe(1);
    wrapper.setState({
        step: 5,
        totalSteps: 5,
        error: false
    });
    expect(wrapper.find(".next").text()).toEqual("Confirm");
    wrapper.find(".next").simulate("click");
    expect(wrapper.state('step')).toBe(5);
    expect(props.onComplete).toBeCalled();
  });
});
