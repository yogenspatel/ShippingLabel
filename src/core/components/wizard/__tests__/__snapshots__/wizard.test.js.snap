// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wizard Component renders correctly has called snapshot with Wizard component 1`] = `
<div
  className="container"
>
  <ShippingProgress
    stepProgress={1}
  />
  <ShippingLabelStep1
    getShippingData={[Function]}
    setShippingData={Object {}}
  />
  <div
    className="card text-center"
  >
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary mr-5 prev"
        onClick={[Function]}
      >
        <span
          className="h4"
        >
          Prev
        </span>
      </button>
      <button
        className="btn btn-primary next"
        onClick={[Function]}
      >
        <span
          className="h4"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</div>
`;
